lighthouse-core/test/lib/traces/trace-json-test.js

```diff
@@ -0,0 +1,77 @@
+ */
+'use strict';
+
+const traceJson = require('../../../lib/traces/trace-json');
+const fs = require('fs');
+const assert = require('assert');
+
+
+/* eslint-env mocha */
+describe('tracejson parser', () => {
+  it('returns preact trace data the same as JSON.parse', (done) => {
+    const filename = '../../fixtures/traces/progressive-app-m60.json';
+    const readStream = fs.createReadStream(__dirname + '/' + filename, {
+      encoding: 'utf-8',
+      // devtools sends traces in 10mb chunks, but this trace is 12MB so we'll do a few chunks
+      highWaterMark: 4 * 1024 * 1024
+    });
+    const parser = new traceJson.TraceParser();
+
+    readStream.on('data', (chunk) => {
+      parser.parseChunk(chunk);
+    });
+    readStream.on('end', () => {
+      const streamedTrace = parser.getTrace();
+      const readTrace = require(filename);
+
+      const streamedLength = streamedTrace.traceEvents.length;
+      const readLength = readTrace.traceEvents.length;
+      assert.equal(streamedLength, readLength);
+      assert.deepStrictEqual(streamedTrace.traceEvents[0], readTrace.traceEvents[0]);
+      assert.deepStrictEqual(
+        streamedTrace.traceEvents[streamedLength -1],
+        readTrace.traceEvents[streamedLength -1]);
+
+      done();
+    });
+  });
+
+
+  it('parses a trace > 256mb (slow)', () => {
+    // FYI: this trace doesn't have a traceEvents property ;)
+    const events = require('../../fixtures/traces/devtools-homepage-w-screenshots-trace.json');
+
+    const stripOuterBrackets = str => str.replace(/^\[/, '').replace(/\]$/, '');
+    const partialEventsStr = events => stripOuterBrackets(JSON.stringify(events));
+
+    const traceEventsStr = partialEventsStr(events.slice(0, events.length-2)) + ',';
+    const parser = new traceJson.TraceParser();
+
+    // read the trace intro
+    parser.parseChunk(`{"traceEvents": [${traceEventsStr}`);
+    let bytesRead = traceEventsStr.length;
+
+    // just keep reading until we've gone over 256 MB
+    while (bytesRead < 256 * 1024 * 1024) {
+      parser.parseChunk(traceEventsStr);
+      bytesRead += traceEventsStr.length;
+    }
+
+    // the CPU Profiler event is last (and big), inject it just once
+    const lastEventStr = partialEventsStr(events.slice(-1));
+    parser.parseChunk(lastEventStr + ']}');
+    bytesRead += lastEventStr.length;
+
+    const streamedTrace = parser.getTrace();
+    // assert.ok(streamedTrace.traceEvents.length > 400 * 1000);
+    assert.ok(bytesRead > 256 * 1024 * 1024, `${bytesRead} bytes read`);
+    // if > 256 MB are read we should have ~480,000 trace events
+    assert.ok(streamedTrace.traceEvents.length > 300 * 1000, 'not >300,000 trace events');
+    // big trace is ~30X larger than the original
+    assert.ok(streamedTrace.traceEvents.length > events.length * 5, 'way more trace events');
+  });
```

Comment:
maybe check that `streamedTrace.traceEvents[events.length-2]` is the same as `events[0]` (I believe those indices are right :). Checks that the test is feeding in the events correctly and that the parser is correctly parsing into at least the second chunk fed in.